libcudacxx/codegen/generators/compare_and_swap.h

```diff
@@ -0,0 +1,175 @@
+//===----------------------------------------------------------------------===//
+//
+// Part of libcu++, the C++ Standard Library for your entire system,
+// under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+// SPDX-FileCopyrightText: Copyright (c) 2024 NVIDIA CORPORATION & AFFILIATES.
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef COMPARED_AND_SWAP_H
+#define COMPARED_AND_SWAP_H
+
+#include <string>
+
+#include "definitions.h"
+#include <fmt/format.h>
+
+static void FormatCompareAndSwap(std::ostream& out)
```

Comment:
Question: Why not `inline`? Isn't that the default for a function in a header?
```suggestion
inline void FormatCompareAndSwap(std::ostream& out)
```

Reply:
Updated to use `inline`.